### PR TITLE
feat: Added samples for navigation with Xaml

### PR DESCRIPTION
**Pull request by vatsashah45**

### CODE REVIEW

**Review by MartinZikmund — APPROVED** [review]

**Review by MartinZikmund — APPROVED** [review]

**Review by MartinZikmund — APPROVED** [review]
Requires README.md, and needs to be added to the list of samples (check some other sample for the general structure, e.g. https://github.com/unoplatform/Uno.Samples/tree/master/UI/ChatGPT)

File: UI/Navigation/Common/shared.targets [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Target Name="CopySharedConfigFiles" BeforeTargets="BeforeBuild">
+    <Message Text="Copying config files to solution" Importance="high" />
+    <Copy SourceFiles="../../Common/global.json" DestinationFolder="../" />
+	<Copy SourceFiles="../../Common/Directory.Build.props" DestinationFolder="../" />
```
Comment (eriklimakc) [inline UI/Navigation/Common/shared.targets:5]:
```suggestion
    <Copy SourceFiles="../../Common/global.json" DestinationFolder="../" />
    <Copy SourceFiles="../../Common/Directory.Build.props" DestinationFolder="../" />
```

Comment (eriklimakc) [reply UI/Navigation/Common/shared.targets:5]: @nickrandolph Myself and @vatsashah45 thought about using MSBuild. Would this be a good approach to have the packages and sdk versions centralised in one place?

File: UI/Navigation/NavigationWithXaml/NavigationWithXaml/NavigationWithXaml.csproj [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    -->
+    <UnoFeatures>
+      Material;
+      Dsp;
+      Hosting;
+      Toolkit;
+      Logging;
+      Mvvm;
+      Navigation;
+      ThemeService;
```
Comment (eriklimakc) [inline UI/Navigation/NavigationWithXaml/NavigationWithXaml/NavigationWithXaml.csproj:43]:
```suggestion
      Material;
      Hosting;
      Toolkit;
      Mvvm;
      Navigation;
```

File: UI/Navigation/NavigationWithXaml/NavigationWithXaml/Presentation/MainPage.xaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            HorizontalAlignment="Center"
+            VerticalAlignment="Center"
+            Spacing="16">
+        <Image Width="150"
+               Height="150"
+               Margin="12"
+               HorizontalAlignment="Center"
+               Source="ms-appx:///Assets/logo.png" />
+        <Button Content="Go to SamplePage"
+                uen:Navigation.Request="Sample" />
```
Comment (eriklimakc) [inline UI/Navigation/NavigationWithXaml/NavigationWithXaml/Presentation/MainPage.xaml:28]: Maybe we could add a simple comment explaining stating that navigation is being triggered by `uen:Navigation.Request` ?

File: UI/Navigation/NavigationWithXaml/NavigationWithXaml/Presentation/MainViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    public MainViewModel(
+        INavigator navigator)
+    {
+        _navigator = navigator;
+        Title = "Main";
+    }
+    public string? Title { get; }
+
+    public ICommand GoToSecond { get; }
```
Comment (eriklimakc) [inline UI/Navigation/NavigationWithXaml/NavigationWithXaml/Presentation/MainViewModel.cs:18]:
```suggestion
```

File: UI/Navigation/NavigationWithXaml/NavigationWithXaml/Presentation/ShellViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+public class ShellViewModel
+{
+    private readonly INavigator _navigator;
+
+    public ShellViewModel(
+        INavigator navigator)
+    {
+        _navigator = navigator;
+        // Add code here to initialize or attach event handlers to singleton services
```
Comment (eriklimakc) [inline UI/Navigation/NavigationWithXaml/NavigationWithXaml/Presentation/ShellViewModel.cs:11]:
```suggestion
```

File: UI/Navigation/NavigationWithXaml/README.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Navigate via XAML
+
+NavigateWithXAML is a sample Uno Platform application demonstrating navigating between pages using the `uen:Navigation.Request`.
```
Comment (eriklimakc) [inline UI/Navigation/NavigationWithXaml/README.md:3]:
```suggestion
**NavigateWithXAML** is a sample Uno Platform application demonstrating how to navigate between pages using the Uno.Extensions Navigation.Request attached property in XAML.
```

We should also point people to the docs: https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Navigation/HowTo-NavigateInXAML.html

File: UI/Navigation/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
+# Uno Platform Navigation Samples
+
+This repository provides simple, to-the-point code [samples](#samples) for the [Uno Platform Navigation](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Navigation/NavigationOverview.html).
```
Comment (eriklimakc) [inline UI/Navigation/README.md:3]:
```suggestion
This repository provides simple, to-the-point code samples for the [Uno Platform Navigation](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Navigation/NavigationOverview.html).
```

File: UI/Navigation/README.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+## What is the Uno Platform?
+
+The Uno Platform is an Open-source platform for building single codebase native mobile, web, desktop, and embedded apps quickly.
+
+It allows C# and WinUI XAML and/or C# code to run on all target platforms while allowing you control of every pixel. It comes with support for Fluent, Material, and Cupertino design systems out of the box. Uno Platform implements a growing number of the WinRT and WinUI APIs, such as **Microsoft.UI.Xaml**, to enable WinUI applications to run on all platforms with native performance.
+
+Use the WinUI tooling from Windows in [Visual Studio](https://www.visualstudio.com/), such as [XAML Hot Reload](https://learn.microsoft.com/visualstudio/xaml-tools/xaml-hot-reload?view=vs-2019) and [C# Hot Reload](https://learn.microsoft.com/visualstudio/debugger/hot-reload), build your application as much as possible on Windows, then validate that your application runs on iOS, Android, macOS, WebAssembly and Desktop.
+
+Visit [our documentation](https://aka.platform.uno/uno-docs-intro) for more details.
```
Comment (eriklimakc) [inline UI/Navigation/README.md:28]:
```suggestion
```

File: UI/Navigation/README.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+## Samples
+
+See a complete list of code samples [here](https://aka.platform.uno/counter-tutorial) or [documentation](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Navigation/NavigationOverview.html) in the official Uno Platform documentation.
```
Comment (eriklimakc) [inline UI/Navigation/README.md:32]:
```suggestion

1. [Navigate with Xaml]()
2. [Navigate via Code]()
...

```

We should list the Navigation samples here.

File: UI/Navigation/README.md [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+### Uno Platform Samples - Issues
+If you encounter any issues with these samples above, please open an issue [here](https://github.com/unoplatform/uno/issues).
```
Comment (eriklimakc) [inline UI/Navigation/README.md:35]:
```suggestion
If you encounter any issues with these samples above, please open an issue [here](https://github.com/unoplatform/Uno.Samples/issues/new).
```

File: UI/Navigation/README.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+## Have other questions? Feature requests? Issues?
+
+Make sure to visit our [FAQ](https://aka.platform.uno/uno-faq), [create an issue](https://github.com/unoplatform/uno/issues), [open a GitHub Discussion](https://github.com/unoplatform/uno/discussions) or visit our [Discord Server](https://platform.uno/uno-discord) - where our engineering team and community will be able to help you.
+
+## Contributors
+Thanks go to these wonderful people (List made with [contrib.rocks](https://contrib.rocks)):
+
+[![Uno.Samples Contributors](https://contrib.rocks/image?repo=unoplatform/Uno.Samples)](https://github.com/unoplatform/Uno.Samples/graphs/contributors)
+
+💖 Thank you.
```
Comment (eriklimakc) [inline UI/Navigation/README.md:46]:
```suggestion
```

**Review by eriklimakc — COMMENTED** [review]
We should also add a link to the navigation folder here: https://github.com/unoplatform/Uno.Samples/blob/master/doc/samples.md

**Comment by eriklimakc** [comment]
Closing this on behalf of https://github.com/unoplatform/Uno.Samples/pull/777

**Pull request closed without merging**